sql/localsql/nipost/poet_registration.go

```diff
@@ -37,38 +39,62 @@ func AddPoetRegistration(
 	return nil
 }
 
-func PoetRegistrationCount(db sql.Executor, nodeID types.NodeID) (int, error) {
-	var count int
+func ClearPoetRegistrations(db sql.Executor, nodeID types.NodeID) error {
 	enc := func(stmt *sql.Statement) {
 		stmt.BindBytes(1, nodeID.Bytes())
 	}
-	dec := func(stmt *sql.Statement) bool {
-		count = int(stmt.ColumnInt64(0))
-		return true
-	}
-	query := `select count(*) from poet_registration where id = ?1;`
-	_, err := db.Exec(query, enc, dec)
-	if err != nil {
-		return 0, fmt.Errorf("get poet registration count for node id %s: %w", nodeID.ShortString(), err)
+	if _, err := db.Exec(`delete from poet_registration where id = ?1;`, enc, nil); err != nil {
+		return fmt.Errorf("clear poet registrations for %s: %w", nodeID.ShortString(), err)
 	}
-	return count, nil
+	return nil
 }
 
-func ClearPoetRegistrations(db sql.Executor, nodeID types.NodeID) error {
+func PoetRegistrationsByNodeId(db sql.Executor, nodeID types.NodeID) ([]PoETRegistration, error) {
+	var registrations []PoETRegistration
+
 	enc := func(stmt *sql.Statement) {
 		stmt.BindBytes(1, nodeID.Bytes())
 	}
-	if _, err := db.Exec(`delete from poet_registration where id = ?1;`, enc, nil); err != nil {
-		return fmt.Errorf("clear poet registrations for %s: %w", nodeID.ShortString(), err)
+
+	dec := func(stmt *sql.Statement) bool {
+		registration := PoETRegistration{
+			Address:  stmt.ColumnText(1),
+			RoundID:  stmt.ColumnText(2),
+			RoundEnd: time.Unix(stmt.ColumnInt64(3), 0),
+		}
+		stmt.ColumnBytes(0, registration.ChallengeHash[:])
+		registrations = append(registrations, registration)
+		return true
 	}
-	return nil
+
+	query := `SELECT hash, address, round_id, round_end FROM poet_registration WHERE id = ?1;`
+
+	_, err := db.Exec(query, enc, dec)
+	if err != nil {
+		return nil, fmt.Errorf("get poet registrations for node id %s: %w", nodeID.ShortString(), err)
+	}
+
+	return registrations, nil
 }
 
-func PoetRegistrations(db sql.Executor, nodeID types.NodeID) ([]PoETRegistration, error) {
+func PoetRegistrationsByNodeIdAndAddresses(
```

Comment:
I think this function wouldn't be needed at all if the nipost builder extended the `existingRegistrations` with the new successful registrations "in memory".